src/utils/derive_offset_generators.nr

```diff
@@ -488,6 +509,49 @@ where
     println("}}");
 }
 
+unconstrained fn verify_offset_generators<Fq, Curve>(cofactor: Field)
+where
+    Fq: BigNum,
+    Curve: CurveParamsTrait<Fq>,
+{
+    // For each curve, compute the offset generators and verify they match the ones in the curve parameters
+    let seed: [u8; 25] = "bigcurve offset generator".as_bytes();
+
+    // we only need well-defined parameters for a and b, for hash_to_curve to work
+    // (no offset gneerators needed)
+    let g: BigCurve<Fq, Curve> = BigCurve::hash_to_curve(seed);
+
+    let g_init: CurveJ<Fq, Curve> = CurveJ { x: g.x, y: g.y, z: Fq::one(), is_infinity: false };
+    let mut gen = g_init;
+
+    let cofactor_bits: [u1; 128] = cofactor.to_be_bits();
+    let mut it: u32 = 0;
+    for i in 0..128 {
+        if (cofactor_bits[i] == 1) {
+            it = i + 1;
+            break;
+        }
+    }
+
+    for i in it..128 {
+        gen = gen.dbl().0;
+        if (cofactor_bits[i] == 1) {
+            gen = gen.incomplete_add(g_init).0;
+        }
+    }
+    let mut genz = gen.z;
+    genz = genz.__invmod();
+    let zz = genz.__mul(genz);
+    let zzz = zz.__mul(genz);
+    let computed_initial = [gen.x.__mul(zz), gen.y.__mul(zzz)];
+    // Verify against the values in the curve parameters
+    let expected_initial = Curve::offset_generator();
+    println(f"expected_initial: {expected_initial}");
+    println(f"computed_initial: {computed_initial}");
+    assert_eq(computed_initial[0], expected_initial[0], "Initial offset generator x mismatch");
+    assert_eq(computed_initial[1], expected_initial[1], "Initial offset generator y mismatch");
+}
```

Comment:
This should be able to be merged with `compute_and_print_offset_generators` as that function is computing the same thing but is not constraining the actual value being used.